ci/tests/general/file_reader.py

```diff
@@ -0,0 +1,448 @@
+        """
+        :param start: Starting date
+        :type start: str / Datetime
+        :param end: End date
+        :type end: str / Datetime
+        :param search_patterns: What error patterns should be recognized
+        :type search_patterns: List of str
+        :param paths_to_file: List of paths of files that will be searched on all nodes
+        :type paths_to_file: List of str
+        :param debug: Boolean to debug or not
+        :type debug: Boolean
+        :param host: ip of the node
+        :type host: str
+        :param suppress_return: only write to file and not return contents
+        :type suppress_return: Boolean
+        :return: Output of a file as string
+        """
+        if not start:
+            start = (datetime.today() - timedelta(hours=1)).strftime("%Y-%m-%d %H:%M:%S")
+        if not end:
+            end = datetime.today().strftime("%Y-%m-%d %H:%M:%S")
+        # For benchmarking purposes
+        start_time = time.time()
+        # Clear the file cache
+        open(self.FILE_PATH, 'w').close()
+
+        for path_to_file in paths_to_file:
+            # Set some initial values
+            count = 0
+            try:
+                size = os.stat(path_to_file)[ST_SIZE]
+            except OSError:
+                # Prevent not found errors from stopping the logger
+                continue
+            begin_range = 0
+            mid_range = size / 2
+            old_mid_range = mid_range
+            end_range = size
+            pos1 = pos2 = 0
+            # Set the default writing prefix
+            if host:
+                write_prefix = "{0} - {1}: ".format(host, path_to_file)
+            else:
+                write_prefix = "{0}: ".format(path_to_file)
+
+            # If only hours are supplied
+            # test for times to be properly formatted, allow hh:mm or hh:mm:ss
+            p = re.compile(r'(^[2][0-3]|[0-1][0-9]):[0-5][0-9](:[0-5][0-9])?$')
+            if p.match(start) or p.match(end):
+                # Determine Time Range
+                yesterday = date.fromordinal(date.today().toordinal() - 1).strftime("%Y-%m-%d")
+                today = datetime.now().strftime("%Y-%m-%d")
+                now = datetime.now().strftime("%R")
+                if start > now or start > end:
+                    search_start = yesterday
+                else:
+                    search_start = today
+                if end > start > now:
+                    search_end = yesterday
+                else:
+                    search_end = today
+                search_start = self._parse_date(search_start + " " + start)
+                search_end = self._parse_date(search_end + " " + end)
+            else:
+                # Set dates
+                search_start = self._parse_date(start)
+                search_end = self._parse_date(end)
+            try:
+                self.opened_file = open(path_to_file, 'r')
```

Comment:
The reasoning behind the normal open is that I access that self.openend_file in different methods aswell to read the lines.
If the file should constantly open and close, it would slow down the reader.